src/components/shared/Entity/ExistingEntityCards/Toolbar.tsx

```diff
@@ -0,0 +1,169 @@
+import {
+    Autocomplete,
+    AutocompleteRenderInputParams,
+    debounce,
+    FilledInputProps,
+    Grid,
+    SxProps,
+    TextField,
+    Theme,
+    Toolbar,
+} from '@mui/material';
+import {
+    semiTransparentBackground,
+    semiTransparentBackgroundIntensified,
+} from 'context/Theme';
+import {
+    ChangeEvent,
+    Dispatch,
+    SetStateAction,
+    SyntheticEvent,
+    useRef,
+} from 'react';
+import { useIntl } from 'react-intl';
+import { SortDirection } from 'types';
+import useConstant from 'use-constant';
+
+interface Props {
+    belowMd: boolean;
+    gridSpacing: number;
+    setSearchQuery: Dispatch<SetStateAction<string | null>>;
+    setSortDirection: Dispatch<SetStateAction<SortDirection>>;
+}
+
+const inputProps: Partial<FilledInputProps> = {
+    disableUnderline: true,
+    sx: {
+        borderRadius: 5,
+        backgroundColor: (theme) =>
+            semiTransparentBackground[theme.palette.mode],
+    },
+};
+
+const toolbarSectionSx: SxProps<Theme> = {
+    '& .MuiInputLabel-root.Mui-focused': {
+        color: (theme) =>
+            theme.palette.mode === 'dark'
+                ? 'primary'
+                : theme.palette.text.secondary,
+    },
+    '& .MuiFilledInput-root:hover, .MuiFilledInput-root.Mui-focused': {
+        backgroundColor: (theme) =>
+            semiTransparentBackgroundIntensified[theme.palette.mode],
+    },
+};
+
+function ExistingEntityCardToolbar({
+    belowMd,
+    gridSpacing,
+    setSearchQuery,
+    setSortDirection,
+}: Props) {
+    const intl = useIntl();
+    const isFiltering = useRef(false);
+
+    const sortDirectionOptions: {
+        direction: SortDirection;
+        message: string;
+    }[] = useConstant(() => [
+        {
+            direction: 'asc',
+            message: intl.formatMessage({
+                id: 'sortDirection.ascending',
+            }),
+        },
+        {
+            direction: 'desc',
+            message: intl.formatMessage({
+                id: 'sortDirection.descending',
+            }),
+        },
+    ]);
+
+    const handlers = {
+        switchSortDirection: (_event: SyntheticEvent, value: string | null) => {
+            const selectedDirection = sortDirectionOptions.find(
+                (option) => option.message === value
+            )?.direction;
+
+            setSortDirection(selectedDirection ? selectedDirection : 'asc');
+        },
+        filterCards: debounce(
+            (event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+                const filterQuery = event.target.value;
+                const hasQuery = Boolean(filterQuery && filterQuery.length > 0);
+
+                isFiltering.current = hasQuery;
+
+                setSearchQuery(hasQuery ? filterQuery : null);
+            },
+            750
+        ),
+    };
+
+    return (
+        <Toolbar
```

Comment:
This repeated stuff taken from `ConnectorToolbar` should have been pulled out and shared between these two.